wasi/src/test/java/com/dylibso/chicory/wasi/WasiPreview1Test.java

```diff
@@ -226,4 +233,43 @@ public void wasiRandom() {
         assertArrayEquals(first, memory.readBytes(0, 123_456));
         assertArrayEquals(second, memory.readBytes(222_222, 87_654));
     }
+
+    @Test
+    public void wasiPositionedWriteWithAppendShouldFail() throws IOException {
+        try (FileSystem fs =
+                Jimfs.newFileSystem(
+                        Configuration.unix().toBuilder().setAttributeViews("unix").build())) {
+
+            var dir = "fs-tests.dir";
+            Path source = new File("../wasi-testsuite/tests/c/testsuite").toPath().resolve(dir);
+            Path target = fs.getPath(dir);
+            copyDirectory(source, target);
+
+            var options = WasiOptions.builder().withDirectory(target.toString(), target).build();
+
+            try (var wasi = WasiPreview1.builder().withOptions(options).build()) {
+                var memory = new Memory(new MemoryLimits(1));
+
+                int fdPtr = 0;
+                int result =
+                        wasi.pathOpen(
+                                memory,
+                                3,
+                                WasiLookupFlags.SYMLINK_FOLLOW,
+                                "pwrite.cleanup",
+                                WasiOpenFlags.CREAT,
+                                0,
+                                0,
+                                WasiFdFlags.APPEND,
+                                fdPtr);
+                assertEquals(WasiErrno.ESUCCESS.value(), result);
+
+                int fd = memory.readInt(fdPtr);
+                assertEquals(4, fd);
```

Comment:
I'm going to forget what is this 4 in a couple of hours :sweat_smile: 
But it's an improvement over my version of it :+1:

Reply:
It's the next open descriptor (0-2 are stdin/out/err, 3 is the pre-opened directory). We don't actually need to assert this, but doesn't hurt to validate anyway.

Reply:
Yeah, I know because I wrote the test with the debugger on my side :sweat_smile: but I'm going to forget about it quickly.